fact_extractor/plugins/unpacking/dlink_shrs/internal/DecryptorLink.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+import hashlib
+import sys
+import binascii
+import pathlib
+
+from Crypto.Cipher import AES
+
+
+class DcryptLink:
+
+    def __init__(self):
+        self.enc_fw = sys.argv[1]
+        self.dec_fw = sys.argv[2]
+        self.dec_key = None
+        self.dataLen1 = None
+        self.dataLen2 = None
+        self.ivec = None
+        self.__setup__()
+
+    def __setup__(self):
+        self.set_decryption_key()
+        self.set_datalen_variables()
+        self.set_ivec()
+
+    @staticmethod
+    def get_expected_sha512_from_fd_at_offset(file, offset, size=0x40):
+        with open(file, 'rb') as f:
+            f.seek(offset, 0)
+            return binascii.hexlify(f.read(size)).decode()
+
+    @staticmethod
+    def calc_sha512_from_fd_at_offset_of_len(file, offset_payload, len_payload, key=None):
+        with open(file, 'rb') as f:
+            f.seek(offset_payload, 0)
+            data = f.read(len_payload)
+            if key:
+                data = data + key
+        return hashlib.sha512(data).hexdigest()
+
+    @staticmethod
+    def verify(calculated, expected):
+        assert expected == calculated, f"\nExpected:\n  {expected}\nGot:\n  {calculated}"
```

Comment:
I think we agreed that assertions should not be used in non-test code

Reply:
@dorpvom would prefer the double quotes throughout the file to be replaced by single ones